tests/PHPStan/Rules/Operators/InvalidBinaryOperationRuleTest.php

```diff
@@ -303,6 +303,528 @@ public function testBug5309(): void
 		$this->analyse([__DIR__ . '/data/bug-5309.php'], []);
 	}
 
+	public function testBcMathNumber(): void
+	{
+		if (PHP_VERSION_ID < 80400) {
+			self::markTestSkipped('Test requires PHP 8.4.');
+		}
+
+		$this->analyse([__DIR__ . '/data/bcmath-number.php'], [
+			[
+				'Binary operation "<<" between BcMath\\Number and BcMath\\Number results in an error.',
```

Comment:
Personally I'd remove this method altogether (testBcMathNumber). We don't need to test all of these situations "results in an error" because presumably it's handled by a single code path. If you still want to test this error, please do it with a separate test file only with a single or a few expressions.

Reply:
You're right - it's sufficiently covered by NSRT. I removed the test case from InvalidBinaryOperationRuleTest and moved the file to NSRT directory.